src/shared/Microsoft.Git.CredentialManager/TerminalMenu.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Git.CredentialManager
+{
+    public class TerminalMenu : IEnumerable<TerminalMenuItem>
+    {
+        private readonly ITerminal _terminal;
+        private readonly IList<TerminalMenuItem> _items = new List<TerminalMenuItem>();
+
+        public TerminalMenu(ITerminal terminal, string title = null)
+        {
+            _terminal = terminal;
+            Title = title;
+        }
+
+        public string Title { get; }
+
+        public void Add(TerminalMenuItem item)
+        {
+            if (item.IsDefault && _items.Any(x => x.IsDefault))
+            {
+                throw new ArgumentException("A default menu item has already been added.");
+            }
+
+            _items.Add(item);
+        }
+
+        public int Show()
+        {
+            TerminalMenuItem[] itemsArr = _items.ToArray();
```

Comment:
This is an artefact of past implementation designs; previously the `Id` of an item was just the index of the item in the array. Therefore I needed the _index_ of the default menu item. Now the `Id` is explicitly set in the `TerminalMenuItem` instead. Will change to avoid `ToArray`.